packages/instantsearch.js/src/widgets/frequently-bought-together/__tests__/frequently-bought-together.test.tsx

```diff
@@ -0,0 +1,570 @@
+      expect(() => {
+        const searchClient = createSearchClient();
+
+        const search = instantsearch({
+          indexName: 'indexName',
+          searchClient,
+        });
+
+        search.addWidgets([
+          frequentlyBoughtTogether({
+            // @ts-expect-error
+            container: undefined,
+          }),
+        ]);
+      }).toThrowErrorMatchingInlineSnapshot(`
+        "The \`container\` option is required.
+
+        See documentation: https://www.algolia.com/doc/api-reference/widgets/frequentlyBoughtTogether/js/"
+      `);
+    });
+
+    test('adds custom CSS classes', async () => {
+      const container = document.createElement('div');
+      const searchClient = createMockedSearchClient();
+
+      const search = instantsearch({
+        indexName: 'indexName',
+        searchClient,
+      });
+
+      search.addWidgets([
+        frequentlyBoughtTogether({
+          container,
+          objectIDs: ['objectID'],
+          cssClasses: {
+            root: 'ROOT',
+            title: 'TITLE',
+            container: 'CONTAINER',
+            list: 'LIST',
+            item: 'ITEM',
+          },
+          templates: {
+            header: 'Frequently bought together',
+          },
+        }),
+      ]);
+
+      search.start();
+
+      await wait(0);
+
+      expect(
+        container.querySelector('.ais-FrequentlyBoughtTogether')
+      ).toHaveClass('ROOT');
+      expect(
+        container.querySelector('.ais-FrequentlyBoughtTogether-title')
+      ).toHaveClass('TITLE');
+      expect(
+        container.querySelector('.ais-FrequentlyBoughtTogether-container')
+      ).toHaveClass('CONTAINER');
+      expect(
+        container.querySelector('.ais-FrequentlyBoughtTogether-list')
+      ).toHaveClass('LIST');
+      expect(
+        container.querySelector('.ais-FrequentlyBoughtTogether-item')
+      ).toHaveClass('ITEM');
+    });
+  });
+
+  describe('templates', () => {
+    test('renders default templates', async () => {
+      const container = document.createElement('div');
+      const containerNoResults = document.createElement('div');
+      const searchClient = createMockedSearchClient();
+
+      const search = instantsearch({ indexName: 'indexName', searchClient });
+
+      search.addWidgets([
+        frequentlyBoughtTogether({
+          container,
+          objectIDs: ['objectID'],
+        }),
+        frequentlyBoughtTogether({
+          container: containerNoResults,
+          objectIDs: ['objectID2'],
+          // Using this to mock an empty response
+          maxRecommendations: 0,
+        }),
+      ]);
+
+      // @MAJOR Once Hogan.js and string-based templates are removed,
+      // `search.start()` can be moved to the test body and the following
+      // assertion can go away.
+      expect(async () => {
+        search.start();
+        await wait(0);
+      }).not.toWarnDev();
+
+      await wait(0);
+
+      expect(container).toMatchInlineSnapshot(`
```

Comment:
This snapshot should render the default header, no?

Reply:
Ah you're right. I misread the current implementation thinking no translations = no header. I'll update this, thanks!